addons/web/static/src/views/fields/formatters.js

```diff
@@ -248,20 +248,20 @@ formatInteger.extractOptions = ({ attrs, options }) => ({
 
 /**
  * Returns a string representing a many2one value. The value is expected to be
- * either `false` or an array in the form [id, display_name]. The returned
- * value will then be the display name of the given value, or an empty string
- * if the value is false.
+ * either `false` or an object containing at least the key "display_name". The
+ * returned value will then be the display name of the given value, or an empty
+ * string if the value is false.
  *
- * @param {[number, string] | false} value
+ * @param {{ display_name: string } | false} value
  * @param {Object} [options] additional options
  * @param {boolean} [options.escape=false] if true, escapes the formatted value
  * @returns {string}
  */
 export function formatMany2one(value, options) {
     if (!value) {
         value = "";
-    } else if (value[1]) {
-        value = value[1];
+    } else if ("display_name" in value ? value.display_name : value[1]) {
```

Comment:
do we have to keep supporting the old api? If so, the docstring should indicate it

Reply:
Yes, we still have pair values from (search_)read rpc. I've adapted the docstring.